File: readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int main(void)
{
    int l = 0;
    string text;
    float L, S;
    text = get_string("text = ");
    L = (float) count_letters(text) / (float) count_words(text) * 100;
    S = (float) count_sentences(text) / (float) count_words(text) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (islower(text[i]) || isupper(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 1;
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            words += 1;
        }
    }
    return words;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i] == '!') || (text[i] == '.') || (text[i] == '?'))
        {
            sentences += 1;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
import cs50

text = cs50.get_string("Text: ")


def count_words(text):
    words = 1
    for i in text:
        if i == " ":
            words += 1
    return words


def count_sentences(text):
    sentences = 0
    for i in text:
        if i in [".", "!", "?"]:
            sentences += 1
    return sentences


def count_letters(text):
    letters = 0
    for i in text:
        if i.isalpha():
            letters += 1
    return letters


def check(index):
    if index < 1:
        print("Before Grade 1")
    elif index > 16:
        print("Grade 16+")
    else:
        print("Grade ", round(index))


L = count_letters(text) / count_words(text) * 100
S = count_sentences(text) / count_words(text) * 100
index = 0.0588 * L - 0.296 * S - 15.8
check(index)
